intake_esm/core.py

```diff
@@ -420,11 +420,8 @@ def search(
 
         if derivedcat_results:
             # Merge results from the main and the derived catalogs
-            esmcat_results = (
-                pd.concat([esmcat_results, *derivedcat_results])
-                .drop_duplicates()
-                .reset_index(drop=True)
-            )
+            esmcat_results = pd.concat([esmcat_results, *derivedcat_results])
+            esmcat_results = esmcat_results[~esmcat_results.astype(str).duplicated()]
```

Comment:
I've got a feeling like this might cause an issue if the elements of the list don't have a uniquely identifying implementation of `__str__`? Would it perhaps be safer to cast list elements to tuples?

It appears `df.drop_duplicates()` relies on hashing the contents of dataframe elements to determine uniqueness.

I'm going to take a look into this in a simple test case.

Reply:
I've double checked & `.astype(str)` uses the `__repr__` of the list elements, not `__str__`, so this should be safe.